commit 1514322fc90090b9a706c221d074ae80b3e1e079
Author: 173210 <root.3.173210@live.com>
Date:   Mon Nov 14 17:32:24 2016 +0900

    [btls] Set CMAKE_POSITION_INDEPENDENT_CODE TRUE for BUILD_DYNAMIC_BTLS
    
    While CMAKE_POSITION_INDEPENDENT_CODE will be automatically set if
    BUILD_SHARED_LIBS is TRUE, it won't if BUILD_DYNAMIC_BTLS is TRUE.
    This change sets CMAKE_POSITION_INDEPENDENT_CODE manually in such a case.

Index: xamarin-pkg-mono/mono/btls/CMakeLists.txt
===================================================================
--- xamarin-pkg-mono.orig/mono/btls/CMakeLists.txt
+++ xamarin-pkg-mono/mono/btls/CMakeLists.txt
@@ -22,7 +22,14 @@ if (NOT "${BTLS_ARCH}" STREQUAL "")
 	message (WARNING "SET ARCH: ${BTLS_ARCH}")
 	set (CMAKE_SYSTEM_PROCESSOR "${BTLS_ARCH}")
 endif ()
-set (C_CXX_FLAGS "-Wall -Wsign-compare -Wmissing-field-initializers -ggdb -fvisibility=hidden")
+if (BUILD_DYNAMIC_BTLS)
+	set (CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+	set (C_CXX_FLAGS "-Wall -Wsign-compare -Wmissing-field-initializers -fPIC -ggdb -fvisibility=hidden")
+elseif (BUILD_SHARED_LIBS)
+	set (C_CXX_FLAGS "-Wall -Wsign-compare -Wmissing-field-initializers -fPIC -ggdb -fvisibility=hidden")
+else ()
+	set (C_CXX_FLAGS "-Wall -Wsign-compare -Wmissing-field-initializers -ggdb -fvisibility=hidden")
+endif()
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${C_CXX_FLAGS} ${BTLS_CFLAGS}")
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${C_CXX_FLAGS} ${BTLS_CFLAGS}")
 set (CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${BTLS_CFLAGS}")
